src/Microsoft.AspNet.Razor/CodeGenerators/Visitors/CSharpTagHelperFieldDeclarationVisitor.cs

```diff
@@ -47,28 +47,29 @@ protected override void Visit(TagHelperChunk chunk)
                     // Need to disable the warning "X is assigned to but never used." for the value buffer since
                     // whether it's used depends on how a TagHelper is used.
                     Writer.WritePragma("warning disable 0414");
-                    WritePrivateField(_tagHelperContext.TagHelperContentTypeName,
-                                      CSharpTagHelperCodeRenderer.StringValueBufferVariableName,
-                                      value: null);
+                    WritePrivateField(
+                        _tagHelperContext.TagHelperContentTypeName,
+                        CSharpTagHelperCodeRenderer.StringValueBufferVariableName,
+                        value: null);
                     Writer.WritePragma("warning restore 0414");
 
-                    WritePrivateField(_tagHelperContext.ExecutionContextTypeName,
-                                      CSharpTagHelperCodeRenderer.ExecutionContextVariableName,
-                                      value: null);
+                    WritePrivateField(
+                        _tagHelperContext.ExecutionContextTypeName,
+                        CSharpTagHelperCodeRenderer.ExecutionContextVariableName,
+                        value: null);
 
-                    Writer
-                        .Write("private ")
-                        .WriteVariableDeclaration(
+                    WritePrivateField(
                         _tagHelperContext.RunnerTypeName,
                         CSharpTagHelperCodeRenderer.RunnerVariableName,
                         value: null);
 
-                    Writer.Write("private ")
-                          .Write(_tagHelperContext.ScopeManagerTypeName)
-                          .Write(" ")
-                          .WriteStartAssignment(CSharpTagHelperCodeRenderer.ScopeManagerVariableName)
-                          .WriteStartNewObject(_tagHelperContext.ScopeManagerTypeName)
-                          .WriteEndMethodInvocation();
+                    Writer
+                        .Write("private global::")
+                        .Write(_tagHelperContext.ScopeManagerTypeName)
```

Comment:
That kind of magic is hard to read. The "global::" additions shouldn't be sprinkled between this level and teensy helpers. Do it at one level, preferably this one since everything is visible.